framework/src/base/NonlinearSystem.C

```diff
@@ -1786,16 +1795,65 @@ NonlinearSystem::computeJacobianInternal(SparseMatrix<Number> &  jacobian)
       {
         _fe_problem.reinitNodeFace(node, boundary_id, 0);
```

Comment:
> just because there is a sideset, it does not mean there will be a MOOSE BC

Right, but doesn't that check, combined with this check:
```
if (_bcs[0].hasNodalBCs(boundary_id) && node->processor_id() == processor_id())
```
guarantee that that nodeset has NodalBCs?